cmd/common/oci.go

```diff
@@ -250,31 +263,93 @@ func NewRunCommand(rootCmd *cobra.Command, runtime runtime.Runtime, hiddenColumn
 
 		ops := make([]operators.DataOperator, 0)
 		for _, op := range operators.GetDataOperators() {
+			if !initializedOperators {
+				// initialize operators if not yet done in PreRun (e.g. when -f was specified)
+				err := op.Init(opGlobalParams[op.Name()])
+				if err != nil {
+					log.Warnf("error initializing operator %s: %v", op.Name(), err)
+					continue
+				}
+			}
 			ops = append(ops, op)
 		}
 		ops = append(ops, clioperator.CLIOperator, combiner.CombinerOperator)
 
 		timeoutDuration := time.Duration(timeoutSeconds) * time.Second
 
-		imageName := args[0]
+		var image string
+		if len(args) > 0 {
+			image = args[0]
+		}
+
+		paramValueMap := make(map[string]string)
+
+		if inFile != "" {
+			f, err := os.Open(inFile)
+			if err != nil {
+				return fmt.Errorf("opening gadget runtime manifest file %s: %w", inFile, err)
+			}
+			specs, err := gadgetmanifest.InstanceSpecsFromReader(f)
+			f.Close()
+			if err != nil {
+				return fmt.Errorf("reading gadget runtime manifest file %s: %w", inFile, err)
+			}
+
+			detachedParam := runtimeParams.Get("detach")
+			isDetach := detachedParam != nil && detachedParam.AsBool()
+
+			if isDetach {
+				// Copy special oci params
+				ociParams.CopyToMap(paramValueMap, "operator.oci.")
+				return runInstanceSpecsDetached(ctx, runtime, specs, runtimeParams, paramValueMap,
```

Comment:
super minor nit: `paramValueMap` is empty at this point, what about passing `ociParams` to `runInstanceSpecsDetached` and moving `ociParams.CopyToMap(paramValueMap, "operator.oci.")` into it?